### PR TITLE
1st version

### PR DESCRIPTION
**Pull request by martinbaerwolff**
OP#1427
OP#1085

### CODE REVIEW

**Review by randy-seng — DISMISSED** [review]

File: .github/workflows/cicd.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+      - name: Install test dependencies
+        run: |
+          pip install pytest
```
Comment (briemla) [inline .github/workflows/cicd.yml:31]: Can we reuse the requirements_dev.txt file to install all test/dev requirements.

File: .github/workflows/cicd.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          pip install pytest-cov
+          pip install flake8
+          pip install mypy
+          python -c 'from OTVision.helpers.machine \
+          import print_has_cuda; print_has_cuda()'
+      - name: Lint codebase
+        run: |
+          flake8 OTVision
+          flake8 tests
```
Comment (briemla) [inline .github/workflows/cicd.yml:40]: Why not `flake8 .` like we do with `mypy`.

File: .github/workflows/linter.yml [hunk 3]
```diff
@@ -1,14 +1,11 @@
+---
 name: super-linter
```
Comment (briemla) [inline .github/workflows/linter.yml:2]: More a general topic: In the cicd action we do most of the linter/type checking stuff. Do we still need super-linter in the OTC repositories?

File: .vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                ".tmp/debug/Testvideo_Cars-Cyclist_FR20_2020-01-01_00-00-00.mp4",
+                "--filetypes",
+                "mp4",
+                "--weights",
+                "yolov5s"
+            ],
+            "windows": {
+                "args": [
+                    "--path",
+                    ".tmp\\debug\\Testvideo_Cars-Cyclist_FR20_2020-01-01_00-00-00.mp4",
```
Comment (briemla) [inline .vscode/launch.json:38]: Is the differentiation between unix and windows necessary or can python use `/` as path separator also on windows?

File: OTVision/detect/yolo.py [hunk 5]
```diff
@@ -0,0 +1,491 @@
...
+            t_get_batch,
+            t_trans,
+            t_det,
+            t_list,
+            t_loop_overhead,
+            len(img_batch),
+        )
+        batch_no += 1
+
+        width = cap.get(3)  # float
```
Comment (briemla) [inline OTVision/detect/yolo.py:119]: Why is this inside the loop?
The variables are used outside the loop and `cap` is defined outside the loop.

File: OTVision/detect/yolo.py [hunk 6]
```diff
@@ -0,0 +1,491 @@
...
+    conf: float,
+    iou: float,
+    size: int,
+    chunksize: int,
+    normalized: bool,
+) -> dict[str, Union[str, int, float]]:
+    return {
+        "detector": "YOLOv5",
+        "weights": str(weights),
+        "conf": conf,
```
Comment (briemla) [inline OTVision/detect/yolo.py:414]: Consider full names, see OP#1524

File: tests/convert/test_convert.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            / Path(h264_ref_video).name,
+        )
+    h264_test_videos = get_files(
+        paths=[test_data_tmp_dir],
+        filetypes=[".h264"],
+    )
+    convert(h264_test_videos)
+
+    # Local function to turn videos into np arrays for comparison
+    def array_from_video(file: Path) -> np.ndarray:
```
Comment (briemla) [inline tests/convert/test_convert.py:43]: Looks like there is no reference to the outer function. Therefore, this method can be moved on top level.

File: tests/helpers/files_test.py [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+def test_replace_filetype_certainOldFiletype(test_dir_with_files: Path) -> None:
+    files = get_files(paths=[test_dir_with_files])
+    new_filetype = ".fancypng"
+    replaced_files = replace_filetype(
+        files=files, old_filetype=".png", new_filetype=new_filetype
+    )
+    for path in replaced_files:
+        if path.is_file():
+            assert path.suffix.lower() != ".png"
```
Comment (briemla) [inline tests/helpers/files_test.py:120]: Why not `== new_filetype`?

File: tests/transform/test_transform.py [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+        test_gpkg_file = (
+            true_gpkg_file.parents[1] / test_data_tmp_dir.name / true_gpkg_file.name
+        )
+        true_utm_tracks_df = gpd.read_file(true_gpkg_file)
+        test_utm_tracks_df = gpd.read_file(test_gpkg_file)
+
+        # Raise error if df's are not equal
+        assert_frame_equal(true_utm_tracks_df, test_utm_tracks_df)
+
+    assert True
```
Comment (briemla) [inline tests/transform/test_transform.py:81]: This test does not check anything. Should be fixed in a future version. Not relevant for MVP.

**Review by briemla — APPROVED** [review]
Comments are meant as starting points for discussions.

**Review by briemla — APPROVED** [review]

**Review by briemla — APPROVED** [review]

**Pull request merged**